### ISSUE TITLE
Borg backup takes very long for first run even when cache exist

### ISSUE THREAD

**Issue by mstimvol** [borgbackup/borg#7767]
## Have you checked borgbackup docs, FAQ, and open GitHub issues?

Yes

## Is this a BUG / ISSUE report or a QUESTION?

Question

#### Your borg version (borg -V).

borg 1.1.16

#### Operating system (distribution) and version.

Debian Bullseye

#### Hardware / network configuration, and filesystems used.

Kubernetes Cluster

#### How much data is handled by borg?

~ 32000 files (2GB)

#### Full borg commandline that lead to the problem (leave away excludes and passwords)

`borg create -s --files-cache=ctime,size ::"$(date +%s)" /var/www/appdata`

## Describe the problem you're observing.

The path `/var/www/appdata` is our volume we want to back up. When we run `borg create` the first time, it's takes 50. When we run `borg create` again, it takes only 3s for the backup. When we destroy/delete/restart the pod and run `borg create` again, it takes 50s again and for all subsequent calls only 3s.

The first time, our borg cache was in the pod. So whenever we destory the pod, the borg cache getting destroyed, too and that's the reason why it takes so long the first time. But now we moved the borg cache to a persistent volume so that it's not getting destroyed, but the problem persists. The first run of `borg create` takes longer - even when the borg cache already exist.

We adjusted the following env variables of borg to move the corresponding borg diretories to the persistent volume:

```
export BORG_BASE_DIR=/borg
export BORG_CACHE_DIR=$BORG_BASE_DIR/.cache
export BORG_CONFIG_DIR=$BORG_BASE_DIR/.config
```

The `/borg` is a separate, persistent volume. So even if we delete/restart the pod, the borg cache is still there.

We're having other systems with 70GB data we need to backup and in this case it takse 20 min for a backup. When we run the backup in the same pod instance again, it takes only 1 minute...

I know that our borg version isn't the latest version, but we're a bit confused that even when there's a valid cache, it takes longer on the first run. Is borg "caching" something somewhere else or what's the reason why it takes to so long the first time?

Edit: When using `borg create -v --list ...` borg tells me on the first run that all files has been modified (each file starts with "M" - according to documentation `‘M’ = regular file, modified`), although nothing has been changed.

**Comment by ThomasWaldmann** [comment]
It's likely due to a mismatch in the attributes used to determine whether a file is unchanged.

See the `--files-cache=ctime,size,inode` option - I suspect maybe the inode numbers in the fs you use are not stable **or** the absolute paths of the input files change (e.g. to different mountpoint).

**Comment by ThomasWaldmann** [comment]
See also `--debug-topic=files_cache` there:

https://borgbackup.readthedocs.io/en/stable/faq.html#why-is-backup-slow-for-me

**Comment by mstimvol** [comment]
Thank @ThomasWaldmann for the feedback. I figured out that whenever the volume is mounted into the pod, the ctime of the file changes. So borg is fast as long as the pod isn't getting destroyed... when the pod is destroyed/restarted by Kubernetes, the volume is mounted again and the ctime changes. According to the documentation `ctime,size,inode` is the default mode for `files-cache`. So I had to change it to use `mtime` instead of `ctime` which now works perfectly.
